platform/ios/src/MGLMapboxEvents.m

```diff
@@ -252,9 +256,13 @@ - (void)pauseOrResumeMetricsCollectionIfRequired {
         [self pauseMetricsCollection];
         return;
     }
+
+    if ([self isLowPowerModeEnabled]) {
+        [self pauseMetricsCollection];
```

Comment:
Yes, we register for [`NSProcessInfoPowerStateDidChangeNotification`](https://developer.apple.com/reference/foundation/nsprocessinfopowerstatedidchangenotification?language=objc), which is sent when Low Power Mode is enabled or disabled. See [line 215](https://github.com/mapbox/mapbox-gl-native/pull/5369/files#diff-6e8e31cd0ffe52aeda9c8b2671bcea19R215).

When we receive this notification, we call `-pauseOrResumeMetricsCollectionIfRequired` and revalidate whether or not telemetry should be enabled. After b9adf31, the LPM check occurs in `+isEnabled`.